packages/hardhat-core/test/internal/hardhat-network/provider/modules/eth/methods/getTransactionReceipt.ts

```diff
@@ -51,6 +54,63 @@ describe("Eth module", function () {
           assert.isNull(receipt);
         });
 
+        it("should return the right tx index and gas used", async function () {
+          const firstBlock = await getFirstBlock();
+          const contractAddress = await deployContract(
+            this.provider,
+            `0x${EXAMPLE_CONTRACT.bytecode.object}`
+          );
+
+          await this.provider.send("evm_setAutomine", [false]);
+
+          const txHash1 = await this.provider.send("eth_sendTransaction", [
+            {
+              to: contractAddress,
+              from: DEFAULT_ACCOUNTS_ADDRESSES[0],
+              data: `${EXAMPLE_CONTRACT.selectors.modifiesState}000000000000000000000000000000000000000000000000000000000000000a`,
+              gas: numberToRpcQuantity(150_000),
+            },
+          ]);
+          const txHash2 = await this.provider.send("eth_sendTransaction", [
+            {
+              to: contractAddress,
+              from: DEFAULT_ACCOUNTS_ADDRESSES[0],
+              data: `${EXAMPLE_CONTRACT.selectors.modifiesState}000000000000000000000000000000000000000000000000000000000000000a`,
+              gas: numberToRpcQuantity(150_000),
+            },
+          ]);
+
+          await this.provider.send("evm_mine", []);
+          const block: RpcBlockOutput = await this.provider.send(
+            "eth_getBlockByNumber",
+            [numberToRpcQuantity(firstBlock + 2), false]
+          );
+
+          assert.equal(block.transactions.length, 2);
+
+          const receipt1: RpcReceiptOutput = await this.provider.send(
+            "eth_getTransactionReceipt",
+            [txHash1]
+          );
+          const receipt2: RpcReceiptOutput = await this.provider.send(
+            "eth_getTransactionReceipt",
+            [txHash2]
+          );
+
+          assert.equal(rpcQuantityToNumber(receipt1.gasUsed), 44945);
```

Comment:
this is a bit nitpicky, but i think checking the exact number of gas here will lead to this tests breaking in the future (e.g. after a new hardfork). I'd test that these pseudocode conditions hold:

```js
receipt1.gasUsed === receipt1.cumulativeGasUsed
receipt2.cumulativeGasUsed === receipt1.cumulativeGasUsed + receipt2.gasUsed
```

Reply:
That's actually one of the things I modified from @ngotchac original code. I did it like this because I like to have the less amount of logic possible in a test, even if that means having to update some values after a harfork. But I'll change it back, no problem.